### PR TITLE
Clickhouse GitHub activity

### PR DESCRIPTION
**Pull request by IvanIvanoff**
#### Summary

[//]: # (#### Related PRs)

[//]: # (#### Additional deploy notes)

[//]: # (#### Screenshots)

### CODE REVIEW

File: config/dev.exs [hunk 1]
```diff
@@ -68,14 +68,14 @@ config :sanbase, Sanbase.TimescaleRepo,
...
   loggers: [Ecto.LogEntry],
-  hostname: "clickhouse",
+  hostname: "clickhouse.default.svc.cluster.local",
   port: 8123,
   database: "default",
   username: "default",
   password: "",
   pool_timeout: 60_000,
   timeout: 60_000,
-  pool_size: {:system, "CLICKHOUSE_POOL_SIZE", "30"}
```
Comment (tspenov) [inline config/dev.exs]: why the pool size is down to 3 ?

Comment (IvanIvanoff) [reply config/dev.exs]: This is only for dev. This pool size means that all 30 connections are opened all the time. Usually you connect to staging and when 2-3 people use it, stage has 100 additional connections.

Comment (tspenov) [reply config/dev.exs]: ups didn't see that its dev

File: lib/sanbase/clickhouse/github.ex [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        UNION ALL
+
+        SELECT toDateTime(intDiv(toUInt32(dt), ?1) * ?1) as time, count(e) as event
+          FROM (
+            SELECT any(event) as e, dt
+            FROM #{@table}
+            PREWHERE owner = ?3
+            AND dt >= toDateTime(?4)
+            AND dt <= toDateTime(?5)
+            AND event NOT in (?6)
```
Comment (valo) [inline lib/sanbase/clickhouse/github.ex]: Do we really want to change the GH activity metric directly? I am bit cautious to change the calculations without examining some comparison between the 2 metrics. May be we can add a new query for the new calculations and provide a side by side comparison? We can also extend the medium article we posted with the extended research.

Comment (valo) [reply lib/sanbase/clickhouse/github.ex]: Oh, I see now. We are actually adding a new query :)

File: lib/sanbase_web/graphql/schema.ex [hunk 3]
```diff
@@ -201,6 +201,22 @@ defmodule SanbaseWeb.Graphql.Schema do
...
+    @desc ~s"""
+    test
+    """
+    field :github_activity2, list_of(:activity_point) do
+      arg(:slug, :string)
+      arg(:from, non_null(:datetime))
+      arg(:to, non_null(:datetime))
+      arg(:interval, :string)
+      arg(:transform, :string, default_value: "None")
+      arg(:moving_average_interval_base, :string, default_value: "1w")
```
Comment (valo) [inline lib/sanbase_web/graphql/schema.ex]: I think this argument is not used right now.

File: lib/sanbase/clickhouse/github.ex (hunk 2 above)
Comment (IvanIvanoff) [reply lib/sanbase/clickhouse/github.ex]: Yes, it's a new query. I think it's worth renaming it to `devActivity`. This has two advantages:
- Implies that it's for development and excluding talking
- Abstracts away from Github. Who knows, we may add gitlab at some time.

File: lib/sanbase_web/graphql/schema.ex (hunk 3 above)
Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/schema.ex]: I plan to implement in Elixir as it's not present in ClickHouse.

File: lib/sanbase/clickhouse/github.ex [hunk 4]
```diff
@@ -111,4 +127,32 @@ defmodule Sanbase.Clickhouse.Github do
...
+      end)
+
+    {:ok, result}
+  end
+
+  defp average([]), do: nil
+
+  defp average(l) when is_list(l) do
+    values = Enum.map(l, fn %{activity: da} -> da end)
+    %{datetime: datetime} = List.last(l)
```
Comment (valo) [inline lib/sanbase/clickhouse/github.ex]: Should we take the first or the last datetime of the group for each SMA group?

File: lib/sanbase_web/graphql/schema.ex (hunk 3 above)
Comment (IvanIvanoff) [reply lib/sanbase_web/graphql/schema.ex]: It's now implemented.

File: lib/sanbase/clickhouse/github.ex (hunk 4 above)
Comment (valo) [reply lib/sanbase/clickhouse/github.ex]: Makes sense

**Pull request merged**